pkg/rancher-desktop/preload/extensions.ts

```diff
@@ -595,6 +642,71 @@ class Client implements v1.DockerDesktopClient {
         };
       });
     },
+    subscribeToEvents: (options: DockerEventSubscriptionOptions = {}, callback: DockerEventCallback): DockerEventSubscription => {
+      const eventArgs = ['events', '--format', '{{json .}}'];
+
+      if (options.filters) {
+        if (options.filters.type) {
+          options.filters.type.forEach(type => {
+            eventArgs.push('--filter', `type=${ type }`);
+          });
+        }
+        if (options.filters.event) {
+          options.filters.event.forEach(event => {
+            eventArgs.push('--filter', `event=${ event }`);
+          });
+        }
+        if (options.filters.container) {
+          options.filters.container.forEach(container => {
+            eventArgs.push('--filter', `container=${ container }`);
+          });
+        }
+        if (options.filters.label) {
+          options.filters.label.forEach(label => {
+            eventArgs.push('--filter', `label=${ label }`);
+          });
+        }
+      }
+
+      const subscriptionId = `events-${ Date.now() }-${ Math.random().toString(36).substring(2, 11) }`;
+
+      const eventProcess = this.docker.cli.exec('system', eventArgs, {
+        stream: {
+          onOutput: (data: { stdout?: string; stderr?: string }) => {
```

Comment:
Destructured assignment here makes it look a lot nicer:
```suggestion
          onOutput: ({ stdout, stderr }) => {
```
Because `docker.cli.exec` is already correctly typed, it knows that `stdout` and `stderr` are both `string | undefined`.